File: dsa/python/ProcessString.py
class Solution(object):
    def processStr(self, s):
        result = ""
        for ch in s:
            if(ord(ch) >= 97 and ord(ch) <= 122):
                result+=ch
            if(ch == '*' and len(result) != 0):
                result = result[:-1]
            if(ch == '#' and len(result) != 0):
                result += result
            if(ch == '%' and len(result) > 1):
                result = result[::-1] 
        return result
      
s1 = Solution()
s = "a#b%*"
print(f"Original String : {s}")
print(f"After Processing String : {s1.processStr(s)}")

File: dsa/cpp/ProcessString.cpp
# include <iostream>
# include <string>
# include <algorithm>
using namespace std;

class Solution {
public:
    string processStr(string s) {
        string result = "";
        int lenght = s.length();
        for(int i=0; i<lenght; i++) {
            if(s[i] >= 'a' && s[i] <= 'z') {
                result += s[i];
            }
            if(s[i] == '*' && !result.empty()) {
                result.pop_back();
            }
            if(s[i] == '#' && !result.empty()) {
                result += result;
            } 
            if(s[i] == '%' && result.length() > 1) {
                reverse(result.begin(), result.end());
            }
        }
        return result;
    }
};


int main () {
    Solution solver;
    string s = "a#b%*";
    cout << "Original String is : " << s << endl;
    cout << "Processed String is : " << solver.processStr(s) << endl;

    return 0;
}